profitbricks-rest/src/test/java/org/apache/jclouds/profitbricks/rest/features/SnapshotApiLiveTest.java

```diff
@@ -0,0 +1,165 @@
+import org.apache.jclouds.profitbricks.rest.internal.BaseProfitBricksLiveTest;
+import static org.testng.Assert.assertEquals;
+import org.testng.annotations.Test;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertNotNull;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+
+@Test(groups = "live", testName = "SnapshotApiLiveTest")
+public class SnapshotApiLiveTest extends BaseProfitBricksLiveTest {
+   
+   DataCenter dataCenter;
+   Volume testVolume;
+   Snapshot testSnapshot;
+   
+   @BeforeClass
+   public void setupTest() {
+      dataCenter = createDataCenter();
+      assertDataCenterAvailable(dataCenter);
+
+      testVolume = api.volumeApi().createVolume(
+         Volume.Request.creatingBuilder()
+         .dataCenterId(dataCenter.id())
+         .name("jclouds-volume")
+         .size(3)
+         .licenceType(LicenceType.LINUX)
+         .build()
+      );
+
+      assertNotNull(testVolume);
+      assertVolumeAvailable(testVolume);
+      
+      testSnapshot = api.volumeApi().createSnapshot(
+         Volume.Request.createSnapshotBuilder()
+            .dataCenterId(testVolume.dataCenterId())
+            .volumeId(testVolume.id())
+            .name("test-snapshot")
+            .description("snapshot desc...")
+            .build());
+
+      assertSnapshotAvailable(testSnapshot);
+   }
+
+   @AfterClass(alwaysRun = true)
+   public void teardownTest() {
+      if (dataCenter != null)
+         deleteDataCenter(dataCenter.id());
+   }
+   
+   @Test
+   public void testList() {
+      List<Snapshot> snapshots = snapshotApi().getList();
+
+      assertNotNull(snapshots);
+      assertFalse(snapshots.isEmpty());
+   }
+
+   @Test
+   public void testGetSnapshot() {
+      Snapshot snapshot = snapshotApi().getSnapshot(testSnapshot.id());
+
+      assertNotNull(snapshot);
+      assertEquals(snapshot.id(), testSnapshot.id());
+      assertEquals(snapshot.properties().name(), "test-snapshot");
+   }  
+   
+   @Test(dependsOnMethods = "testGetSnapshot")
+   public void testUpdateSnapshot() {
+      Snapshot snapshot = snapshotApi().updateSnapshot(
+         Snapshot.Request.updatingBuilder()
+         .id(testSnapshot.id())
+         .name("test-snapshot new name")
+         .build()
+      );
+
+      assertVolumeAvailable(testVolume);
+      assertEquals(snapshot.properties().name(), "test-snapshot new name");
+   }
+   
+   @Test(dependsOnMethods = "testUpdateSnapshot")
+   public void testDeleteSnapshot() {
+      api.volumeApi().deleteVolume(testVolume.dataCenterId(), testVolume.id());
+      assertVolumeRemoved(testVolume);
+      snapshotApi().deleteSnapshot(testSnapshot.id());
+      assertSnapshotRemoved(testSnapshot);
+   }
+   
+   private SnapshotApi snapshotApi() {
+      return api.snapshotApi();
+   }
+   
+   private void assertVolumeAvailable(Volume volume) {
+      assertRandom(new Predicate<VolumeRef>() {
```

Comment:
Just seen this... Would it be more convenient for this method to be named `assertPredicate`? The current name seems to mean that it will perform arbitrary assertions...